vector/src/main/java/im/vector/app/features/signout/soft/SoftLogoutViewModel.kt

```diff
@@ -56,15 +57,23 @@ class SoftLogoutViewModel @AssistedInject constructor(
     companion object : MavericksViewModelFactory<SoftLogoutViewModel, SoftLogoutViewState> by hiltMavericksViewModelFactory() {
 
         override fun initialState(viewModelContext: ViewModelContext): SoftLogoutViewState? {
-            val activity: SoftLogoutActivity = (viewModelContext as ActivityViewModelContext).activity()
-            val userId = activity.session.myUserId
-            return SoftLogoutViewState(
-                    homeServerUrl = activity.session.sessionParams.homeServerUrl,
-                    userId = userId,
-                    deviceId = activity.session.sessionParams.deviceId ?: "",
-                    userDisplayName = activity.session.getUser(userId)?.displayName ?: userId,
-                    hasUnsavedKeys = activity.session.hasUnsavedKeys()
-            )
+            val sessionHolder = EntryPoints.get(viewModelContext.app(), SingletonEntryPoint::class.java)
+                    .activeSessionHolder()
+
+            return if (sessionHolder.hasActiveSession()) {
+                val session = sessionHolder.getActiveSession()
+                val userId = session.myUserId
+
+                SoftLogoutViewState(
+                        homeServerUrl = session.sessionParams.homeServerUrl,
+                        userId = userId,
+                        deviceId = session.sessionParams.deviceId.orEmpty(),
+                        userDisplayName = session.getUser(userId)?.displayName ?: userId,
+                        hasUnsavedKeys = session.hasUnsavedKeys()
+                )
+            } else {
+                null
```

Comment:
In fact, Mavericks handles null viewState: the doc says: 
`Returns:
the initial state. If null, the state class constructors will be used for initial state creation.`
But I fixed it since there is no default constructor with empty arguments. With null it crashes. Thanks! 
However, here `SoftLogoutViewModel` cannot be instantiated if there is no active `Session` so this `else` branch should never be reached... And this is why, I removed it from `FtueAuthWebFragment`.